src/lib/CollisionPrevention/CollisionPrevention.cpp

```diff
@@ -0,0 +1,189 @@
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file CollisionPrevention.cpp
+ * CollisionPrevention controller.
+ *
+ */
+
+#include <CollisionPrevention/CollisionPrevention.hpp>
+
+
+CollisionPrevention::CollisionPrevention() :
+	ModuleParams(nullptr)
+{
+
+}
+
+CollisionPrevention::~CollisionPrevention()
+{
+	//unadvertise publishers
+	if (_constraints_pub != nullptr) {
+		orb_unadvertise(_constraints_pub);
+	}
+
+	if (_mavlink_log_pub != nullptr) {
+		orb_unadvertise(_mavlink_log_pub);
+	}
+}
+
+bool CollisionPrevention::initializeSubscriptions(SubscriptionArray &subscription_array)
+{
+	if (!subscription_array.get(ORB_ID(obstacle_distance), _sub_obstacle_distance)) {
+		return false;
+	}
+
+	return true;
+}
+
+void CollisionPrevention::reset_constraints()
+{
+
+	_move_constraints_x_normalized.zero();  //normalized constraint in x-direction
+	_move_constraints_y_normalized.zero();  //normalized constraint in y-direction
+
+	_move_constraints_x.zero();  //constraint in x-direction
+	_move_constraints_y.zero();  //constraint in y-direction
+}
+
+void CollisionPrevention::publish_constraints(const Vector2f &original_setpoint, const Vector2f &adapted_setpoint)
+{
+
+	collision_constraints_s	constraints;	/**< collision constraints message */
+
+	//fill in values
+	constraints.timestamp = hrt_absolute_time();
+	constraints.constraints_normalized_x[0] = _move_constraints_x_normalized(0);
+	constraints.constraints_normalized_x[1] = _move_constraints_x_normalized(1);
+	constraints.constraints_normalized_y[0] = _move_constraints_y_normalized(0);
+	constraints.constraints_normalized_y[1] = _move_constraints_y_normalized(1);
+
+	constraints.original_setpoint[0] = original_setpoint(0);
+	constraints.original_setpoint[1] = original_setpoint(1);
+	constraints.adapted_setpoint[0] = adapted_setpoint(0);
+	constraints.adapted_setpoint[1] = adapted_setpoint(1);
+
+	// publish constraints
+	if (_constraints_pub != nullptr) {
+		orb_publish(ORB_ID(collision_constraints), _constraints_pub, &constraints);
+
+	} else {
+		_constraints_pub = orb_advertise(ORB_ID(collision_constraints), &constraints);
+	}
+}
+
+void CollisionPrevention::update_range_constraints()
+{
+	obstacle_distance_s obstacle_distance = _sub_obstacle_distance->get();
+
+	if (hrt_elapsed_time(&obstacle_distance.timestamp) < RANGE_STREAM_TIMEOUT_US) {
+		float max_detection_distance = obstacle_distance.max_distance / 100.0f; //convert to meters
+
+		for (int i = 0; i < 72; i++) {
```

Comment:
Please use `sizeof(obstable_distance::distances)/sizeof(obstable_distance::distances[0])` instead of 72.